mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -556,6 +579,28 @@ class SparseVector(
       i += 1
     }
   }
+
+  override def hashCode(): Int = {
```

Comment:
Should SparseVector use the first 16 non-zero values instead of the first 16?  For sparse data, it would not be surprising for a bunch of vectors to have all-zeros in the first 16 elements.